File: BOJ/python3/2775.py
'''
부녀회장이 될테야
https://www.acmicpc.net/problem/2775
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력

T = int(input())

# 아파트 전체 인원
A = []
A.append([n for n in range(1, 15)])
for k in range(1, 15):
    A.append([0 for _ in range(14)])
    i = 0
    for n in range(14):
        i += A[k-1][n]
        A[k][n] = i


for t in range(T):
    k = int(input())  # 층수
    n = int(input())  # 호수

    print(A[k][n-1])


# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/2775.js
/*
부녀회장이 될테야
boj.kr/2775
 */
const start = new Date().getTime();

const fs = require("fs");
const input = fs.readFileSync("input.txt").toString().split("\n");
//const input = fs.readFileSync("/dev/stdin").toString().split("\n");

const A = Array.from(Array(15), (v, r) =>
  Array.from({ length: 15 }, (v, c) => c)
);
for (let k = 1; k < 15; k++) {
  for (let n = 1; n < 15; n++) {
    A[k][n] = A[k][n - 1] + A[k - 1][n];
  }
}

let line = 0;
const T = 1 * input[line++];
let ans = [];
for (let t = 0; t < T; t++) {
  const k = 1 * input[line++];
  const n = 1 * input[line++];
  ans.push(A[k][n]);
}

console.log(ans.join("\n"));

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
